app/services/subscriptions/dates_service.rb

```diff
@@ -117,7 +117,7 @@ def charges_to_datetime
       return unless subscription.started_at
 
       datetime = customer_timezone_shift(compute_charges_to_date, end_of_day: true)
-      datetime = subscription.terminated_at if subscription.terminated_at?(datetime)
+      datetime = subscription.terminated_at if subscription.terminated? && subscription.terminated_at <= datetime
```

Comment:
found a bug about this termination: prorated charges are counted +1 day of proration for terminated subscriptions:
    # TODO: Why it was <= ?
    # the situation that i have:
    # timestamp: 2025-08-31 23:59:59
    # terminated_at: 2025-09-01 00:00:00
    # round makes both of them 2025-09-01 00:00:00

    # TODO: check how it works for charges - no, it does not. We need round to terminate the subscription
    # (CreateInvoiceSubscriptionService should return subscription_terminated for invoiving reason, that's why)
    # we do need the round, but the date_service should not return terminated_at instead of the received date...